### PR TITLE
Applying SDK Fix `Fix Hash issue in 4.1.0`

### PR DESCRIPTION
**Pull request by shaopeng-gh**
This PR demonstrate the change in BinSkim if the SDK fix is applied https://github.com/microsoft/sarif-sdk/pull/2667.
The exact changes for hashes in https://github.com/microsoft/binskim/pull/853 will be reverted to the original hash.

**Update: the sdk fix is in, this is actually applied.**
**Update 2: we reverted the sdk fix, but this PR currently points to the commit that is before revert.**

### CODE REVIEW

**Review by michaelcfanning — CHANGES_REQUESTED** [review]
:clock1:

File: ReleaseHistory.md [hunk 1]
```diff
@@ -14,7 +14,8 @@
 - UEE => eliminate unhandled exceptions in engine
 - NEW => new feature 
 
-## UNRELEASED
+## **v4.1.1** UNRELEASED
```
[the remark this answers is not in the document]
Comment (shaopeng-gh) [reply ReleaseHistory.md:17]: How do you generate your MD%/SHA256 hashes?

----- I use powershell, for example the `gcc.stack_protector.a`

image

[the remark this answers is not in the document]
Comment (shaopeng-gh) [reply ReleaseHistory.md:17]: oh yes I verified which hash is correct with PowerShell before submitted the PR in SARIF SDK.
Before new pipeline change, we had the correct hash.

What accounts for the change in hashing behavior --- hash on the text representation of a binary file will have different value than hash the binary file itself.

These two are the comparison PR:
https://github.com/microsoft/binskim/pull/900
https://github.com/microsoft/binskim/pull/901

**Comment by suvamM** [comment]
@shaopeng-gh I am quite confused by the PR description. The text should be self-contained: for someone with low context, the problem this PR is trying to address, and the proposed fix should be immediately clear.

**Review by suvamM — CHANGES_REQUESTED** [review]
I am unclear what this PR is doing: is it a simple SARIF SDK submodule update, which necessitates changes to the tests? Some more details in the PR description will be helpful.

**Comment by shaopeng-gh** [comment]
> This PR demonstrate the change in BinSkim if the SDK fix is applied [microsoft/sarif-sdk#2667](https://github.com/microsoft/sarif-sdk/pull/2667).
> The exact changes for hashes in #853 will be reverted to the original hash.

Yes! This PR was for the linked PR https://github.com/microsoft/sarif-sdk/pull/2667 
but we now will have a different direction for the fix:
"Yes, to both. You fix the analyze options base. 😊
I have an idea for the other hashing issue."
Let me close this one.

**Pull request closed without merging**